src/rustdoc_gen.rs

```diff
@@ -597,13 +600,56 @@ impl core::fmt::Debug for RustdocFromRegistry {
 
 impl RustdocFromRegistry {
     pub fn new(target_root: &std::path::Path, config: &mut GlobalConfig) -> anyhow::Result<Self> {
-        let mut index = crates_index::Index::new_cargo_default()?;
+        let index_url = tame_index::IndexUrl::crates_io(
+            // This is the config root, where .cargo/config.toml configuration files
+            // are crawled to determine if crates.io has been source replaced
+            // <https://doc.rust-lang.org/cargo/reference/source-replacement.html>
+            // if not specified it defaults to the current working directory,
+            // which is the same default that cargo uses, though note this can be
+            // extremely confusing if one can specify the manifest path of the
+            // crate from a different current working directory, though AFAICT
+            // this is not how this binary works
```

Comment:
The `--manifest-path` flag can be used to select a crate from a different working directory, so I don't believe this is accurate. (The plan is to merge `cargo-semver-checks` into cargo itself, so our flags try as much as possible to behave as cargo's own.)

At the point where `RustdocFromRegistry` is constructed, the `--manifest-path` value (if any) is not present anymore. But as one possible solution, we could put the `index_url` value inside `cargo_semver_checks::Check` and then inside `GlobalConfig`, which will make it available throughout. Or if the better thing to carry around is a `ComboIndexCache`, we could do the same with that struct.

Just one thing to note: it's possible to use the binary and library here without an index. For example, if the user has already generated baseline and current rustdoc JSON files and just wants to semver-check them. This shouldn't fail with a network or configuration error on account of the (unused) index.

Reply:
Yah I noticed that trying to debug why a test was failing, I think for now I'll keep this as it is since it will follow the same cargo logic of using configuration based on the current directory rather than the parent of the manifest, but IMO that logic is incredibly confusing if you don't know about it.

Reply:
Indeed, I did not know about it and I am finding it confusing! But I agree, matching what cargo does is the right call here.